x-pack/plugin/logsdb/src/javaRestTest/java/org/elasticsearch/xpack/logsdb/qa/matchers/source/SourceMatcher.java

```diff
@@ -104,7 +104,7 @@ private MatchResult compareSource(Map<String, List<Object>> actual, Map<String,
             // See #111916.
```

Comment:
As mentioned in the comment ideally we should remove this logic completely and have a stricter definition of "source matches exactly". It is not easy to do currently because we need to know things like value of `dynamic` of parent objects of dynamically mapped fields and we don't have a good way to obtain this information.